src/csi/mod.rs

```diff
@@ -294,121 +293,120 @@ mod test {
+            "test-snapshot-name".to_owned(), //snap_name,
+            snap_id.to_owned(),              //snap_id,
+            vol_id.to_owned(),
+            meta_data.get_node_id().to_owned(),
+            meta_data.get_snapshot_path(snap_id),
+            std::time::SystemTime::now(),
+            0, // size_bytes,
+        );
+        let add_snap_res = meta_data.add_snapshot_meta_data(snap_id, &snapshot).await;
+        assert!(
+            add_snap_res.is_ok(),
+            "failed to add new snapshot meta data to etcd"
+        );
+        let get_snap_by_name_res = meta_data
+            .get_snapshot_by_name(&snapshot.snap_name)
+            .await
+            .with_context(|| format!("failed to find snapshot by name={}", snapshot.snap_name,))?;
+        assert_eq!(
+            get_snap_by_name_res.snap_name, snapshot.snap_name,
+            "snapshot name not match"
+        );
 
-            let get_snap_by_src_vol_id_res = meta_data
-                .get_snapshot_by_src_volume_id(&snapshot.vol_id)
-                .await
-                .with_context(|| {
-                    format!(
-                        "failed to find snapshot by source volume ID={}",
-                        snapshot.vol_id,
-                    )
-                })?;
-            assert_eq!(
-                get_snap_by_src_vol_id_res.vol_id, snapshot.vol_id,
-                "snapshot source volume ID not match"
-            );
+        let get_snap_by_src_vol_id_res = meta_data
+            .get_snapshot_by_src_volume_id(&snapshot.vol_id)
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to find snapshot by source volume ID={}",
+                    snapshot.vol_id,
+                )
+            })?;
+        assert_eq!(
+            get_snap_by_src_vol_id_res.vol_id, snapshot.vol_id,
+            "snapshot source volume ID not match"
+        );
 
-            let del_snap_res = meta_data.delete_snapshot_meta_data(snap_id).await?;
-            assert_eq!(
-                del_snap_res.snap_id, snap_id,
-                "deleted snapshot ID not match"
-            );
-            let del_vol_res = meta_data
-                .delete_volume_meta_data(vol_id, meta_data.get_node_id())
-                .await?;
-            assert_eq!(del_vol_res.vol_id, vol_id, "deleted volume ID not match");
-            Ok(())
-        })
+        let del_snap_res = meta_data.delete_snapshot_meta_data(snap_id).await?;
+        assert_eq!(
+            del_snap_res.snap_id, snap_id,
+            "deleted snapshot ID not match"
+        );
+        let del_vol_res = meta_data
+            .delete_volume_meta_data(vol_id, meta_data.get_node_id())
+            .await?;
+        assert_eq!(del_vol_res.vol_id, vol_id, "deleted volume ID not match");
+        Ok(())
     }
 
     fn get_volume_path(vol_id: &str) -> PathBuf {
         Path::new(DATA_DIR).join(vol_id)
     }
 
+    #[allow(clippy::pedantic)]
```

Comment:
what kind of clippy is triggered here?

Reply:
error: this function's return value is unnecessary

Reply:
modified to a lesser range allow